src/Hazelcast.Net/Networking/CloudOptions.cs

```diff
@@ -32,15 +32,14 @@ public CloudOptions()
         /// </summary>
         private CloudOptions(CloudOptions other)
         {
-            Enabled = other.Enabled;
             DiscoveryToken = other.DiscoveryToken;
             UrlBase = other.UrlBase;
         }
 
         /// <summary>
         /// Whether Hazelcast Cloud is enabled.
         /// </summary>
-        public bool Enabled { get; set; }
+        public bool Enabled => !string.IsNullOrWhiteSpace(DiscoveryToken);
```

Comment:
I think we should remove the Enable field. The token field is used for enabling. node.js uses the same idea.

Reply:
ah, will make it internal all right - it's nice to have because in various places we want to know whether it's "enabled" and it's better to have a bool property - but I get it that users don't need to see it